### PR TITLE
Add typescript types

### PR DESCRIPTION
**Pull request by Flechaa**
Adds typescript types based on the ``index.js`` exports.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
+export type Leagues = "little" | "great" | "ultra" | "master";
```
Comment (Mygod) [inline types/index.d.ts]: There shouldn't be a `Leagues` type... In principle they can be arbitrary strings.

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+export interface LRUCache {
+    get(key: string): object;
+    set(key: string, value: object): void;
+}
+
+export interface OhbemOptions {
+    leagues?: Record<string, LeagueOptions | number | null>;
+    levelCaps?: number[];
+    pokemonData?: PokemonData;
```
Comment (Mygod) [inline types/index.d.ts:94]: I think `PokemonData` already has a ts? @TurtIeSocks

**Review by Mygod — COMMENTED** [review]
Thanks for the contribution! I took a quick look and have some questions.

File: types/index.d.ts (hunk 2 above)
Comment (Flechaa) [reply types/index.d.ts:94]: I found something [here](https://github.com/WatWowMap/Pogo-Data-Generator/blob/5608afbb704f3ed30524dd1b86f791d26700aa66/src/typings/dataTypes.ts#L88) but the actual result seems to depend on the template provided.

Comment (Mygod) [reply types/index.d.ts:94]: Yes yours is better. Thanks again! I will take a closer look and merge when I get a chance.

**Review by TurtIeSocks — APPROVED** [review]

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    constructor(options?: OhbemOptions);
+
+    calculateAllRanks(stats: Stats, cpCap: number): Record<string, RankResult[][][]> | null;
+    calculateTopRanks(maxRank: number, pokemonId: number, form?: number, evolution?: number, ivFloor?: number): PvPQueryResult;
+    updatePokemonData(pokemonData: PokemonData): void;
+    queryPvPRank(pokemonId: number, form: number, costume: number, gender: number, attack: number, defense: number, stamina: number, level: number): PvPQueryResult;
+    findBaseStats(pokemonId: number, form?: number, evolution?: number): Pokemon;
+}
+
+export = Ohbem;
```
Comment (Mygod) [inline types/index.d.ts]: Is this statement okay? `TS2309: An export assignment cannot be used in a module with other exported elements.`

File: types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+     * @returns The filtered array containing only capped entries and those whose cap matches the given caps.
+     */
+    static filterLevelCaps(entries: RankEntry[], interestedLevelCaps: number[]): RankEntry[];
+
+    /**
+     * Initialize your overlord Ohbem.
+     */
+    constructor(options?: OhbemOptions);
+
+    calculateAllRanks(stats: Stats, cpCap: number): Record<string, RankResult[][][]> | null;
```
Comment (Mygod) [inline types/index.d.ts]: This return value is not correct, it could be two different values depending on if compact cache is enabled or not.

**Review by Mygod — COMMENTED** [review]
Hi please see some comments above. Also please also help move the documentations from `index.js` to the `index.d.ts` file. :)

**Comment by Mygod** [comment]
Thanks! LGTM now.

**Pull request merged**